### PR TITLE
Don't choose the same map again after re-shuffling

### PR DESCRIPTION
**Pull request by Thomas--S**
Tested and should work fine.

### CODE REVIEW

**Comment by ClobberXD** [comment]
Is this to prevent the last map of the previous cycle from becoming the first in the next cycle?

File: mods/ctf/ctf_map/ctf_map_core/schem_map.lua [hunk 1]
```diff
@@ -214,7 +214,7 @@ math.randomseed(os.time())
 
 -- Fisher-Yates shuffling algorithm, used for shuffling map selection order
 -- Adapted from snippet provided in https://stackoverflow.com/a/35574006
-local function shuffle_maps()
+local function shuffle_maps(not_first)
```
Comment (ClobberXD) [inline mods/ctf/ctf_map/ctf_map_core/schem_map.lua]: Variable name `not_first` is confusing at first read, and a better name would be nice.

File: mods/ctf/ctf_map/ctf_map_core/schem_map.lua [hunk 2]
```diff
@@ -230,6 +230,11 @@ local function shuffle_maps()
 		local j = math.random(i)
 		shuffled_order[i], shuffled_order[j] = shuffled_order[j], shuffled_order[i]
 	end
+	if shuffled_order[1] == not_first then
```
Comment (ClobberXD) [inline mods/ctf/ctf_map/ctf_map_core/schem_map.lua]: A comment above this line explaining what this check is and what this block does, would be very useful.

File: mods/ctf/ctf_map/ctf_map_core/schem_map.lua (hunk 1 above)
Comment (Thomas--S) [reply mods/ctf/ctf_map/ctf_map_core/schem_map.lua]: Any suggestions?

How about `previous_idx`?

**Comment by Thomas--S** [comment]
> Is this to prevent the last map of the previous cycle from becoming the first in the next cycle?

Yes, it is.

File: mods/ctf/ctf_map/ctf_map_core/schem_map.lua (hunk 1 above)
Comment (ClobberXD) [reply mods/ctf/ctf_map/ctf_map_core/schem_map.lua]: Hmmm... maybe `idx_to_avoid`? It's a little too verbose, but one would immediately understand what it is.

**Review by ClobberXD — APPROVED** [review]
LGTM :+1:

**Pull request merged**